problemtools/verifyproblem.py

```diff
@@ -245,11 +215,32 @@ def __init__(self, problem, datadir, parent=None):
             except Exception as e:
                 self.error(e)
                 self.config = {}
-        elif parent is not None:
-            self.config = parent.config.copy()
         else:
             self.config = {}
 
+        # For non-root groups, missing properties are inherited from the parent group
+        if parent:
+            for field, parent_value in parent.config.iteritems():
+                if not field in self.config:
+                    self.config[field] = parent_value
+
+        # Some deprecated properties are inherited from problem config during a transition period
+        problem_grading = problem.config.get('grading')
+        for key in ['accept_score', 'reject_score', 'range']:
+            if key in problem.config.get('grading'):
+                self.config[key] = problem_grading[key]
+
+        problem_on_reject = problem_grading.get('on_reject')
+        if problem_on_reject == 'first_error':
+            self.config['on_reject'] = 'break'
+        if problem_on_reject == 'grade':
+            self.config['on_reject'] = 'continue'
```

Comment:
For these deprecated properties, maybe only take the data from problem.yaml for the root group?  (It should be the same but matches more closely how at least I think about it.)

Also, maybe I'm reading too sloppily but it looks like any value specified in problem.yaml will override the values specified in testdata.yaml?  The problem.yaml values should only be used if there was nothing specified in testdata.yaml.